src/ResourceManager/KeyVault/Commands.KeyVault/Commands/NewAzureKeyVault.cs

```diff
@@ -140,23 +148,43 @@ public override void ExecuteCmdlet()
                     };
                 }
 
+                // may not specify 'false' for either soft-delete or disable-purge; 
+                // the service, as of Mar 2018, automatically converts this parameter to a 'true', ignoring
+                // the value specified by the client. This, however, may cause an unwanted enabling of
+                // soft delete in a scenario like this:
+                //
+                //      $obj = get-vault 
+                //      delete-vault $obj
+                //      $obj | new-vault
+                //
+                // If s/d was not enabled on the original vault, the parameter name will be present on
+                // the input to the new-vault cmdlet, but without a value. if the presence of the parameter
+                // is passed into the input of the new-vault cmdlet, s/d will be enabled on the resulting vault.
+                if (EnableSoftDelete.IsPresent
```

Comment:
I want to prevent the situation where the property is not set on the vault, but displayed as "SoftDelete: false" in the output of a Get cmdlet (and being piped into the New command). I'm guessing that specifying "ValueFromPipelineByPropertyName = false" would prevent that, can you confirm?

Reply:
Yes, that is correct.